bsip-0040.md

```diff
@@ -94,15 +113,17 @@ Stateless assert, all argument types. `Argument value` must be equal to one of v
 Stateless assert, all argument types. `Argument value` must NOT be equal to any of the values in the list.
 
 #### `lt, le, gt, ge`
-Stateless assert, applies to `int` and `string` type arguments. If argument is `string` type, use its length as `argument value`. 
+Stateless assert. Allows explicit type converstion:
+- `int` type: use as is
+- `string` type: user `length` as `argument value`
+- `price` struct `{base = {amount, asset_id}, quote = {amount, asset_id}}`: convert to float price  `argument value = base.amount/quote.amount`
```

Comment:
Then the value need to be a float point. We don't support float points in consensus code. Perhaps you'll say use string? I would say no, it's inefficient.

Perhaps we can support some simple math/expressions, or predefined function?

Reply:
I have removed implicit `price` converstion. If user wants a price restriction he can use `attribute_assert`.